### PR TITLE
refactor: move telemetry interfaces to core interfaces

### PR DESCRIPTION
**Pull request by RishhiB**
In preparation to remove some shared interfaces from common-definitions, I have copied the implementation of this interface from common-definitions to core-interfaces.

### Moved to @fluidframework/core-interfaces

-	interface ILoggingError
-	interface ITaggedTelemetryPropertyType
-	interface ITelemetryBaseEvent
-	interface ITelemetryBaseLogger
-	interface ITelemetryErrorEvent
-	interface ITelemetryGenericEvent
-	interface ITelemetryLogger
-	interface ITelemetryPerformanceEvent
-	interface ITelemetryProperties
-	type TelemetryEventCategory
-	type TelemetryEventPropertyType

Related to [AB#4057](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4057)

### CODE REVIEW

File: common/lib/common-utils/package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
 	},
 	"dependencies": {
 		"@fluidframework/common-definitions": "^0.20.1",
+		"@fluidframework/core-interfaces": "workspace:~",
```
Comment (anthony-murphy) [inline common/lib/common-utils/package.json:79]: this is invalid. this must remain on common-def. you can't reference a client package from common

File: common/lib/common-definitions/CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,21 @@
 
 ## [1.2.0](https://github.com/microsoft/FluidFramework/releases/tag/common-definitions_v1.2.0)
 
-### Deprecated interface
+### Deprecated interfaces and types
 
-The following interface is deprecated in this release. Its replacement has been moved to another package.
+The following interfaces and types are deprecated in this release. Its replacement has been moved to another package.
```
Comment (alexvy86) [inline common/lib/common-definitions/CHANGELOG.md:7]: Nit
```suggestion
The following interfaces and types are deprecated in this release. Their replacements have been moved to another package.
```

File: common/lib/common-utils/src/logger.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {
 	ITelemetryGenericEvent,
 	ITelemetryLogger,
 	ITelemetryPerformanceEvent,
-} from "@fluidframework/common-definitions";
+} from "@fluidframework/core-interfaces";
```
Comment (alexvy86) [inline common/lib/common-utils/src/logger.ts:13]: Tying to Tony's comment, this import needs to keep using common-definitions for now, until/if these implementations move somewhere else.

**Review by tylerbutler — APPROVED** [review]
Looks good to me. I'd feel better if someone other than me also approves also, though. :smile:

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**